src/odb/src/db/dbObstruction.cpp

```diff
@@ -392,6 +410,15 @@ void dbObstruction::destroy(dbObstruction* obstruction)
 {
   _dbObstruction* obs = (_dbObstruction*) obstruction;
   _dbBlock* block = (_dbBlock*) obs->getOwner();
+
+  if (obstruction->isSystemReserved()) {
+    utl::Logger* logger = block->getLogger();
+    logger->critical(
+        utl::ODB,
+        1111,
+        "You cannot delete a system created obstruction (isSystemReserved).");
+  }
+
```

Comment:
I would be okay with this being a silent error. The usecase I would worry about is one where a user gets all the obstructions and iterates over them to destroy them, that would just throw this critical error, which shouldn't be needed. Same for blockage

Reply:
I agree it should be an error but why silent?

Reply:
This follows the idiom of dbInst DONT_TOUCH instances. It feels like we're protecting the integrity of the app. We could make it warning, and just not delete them if you want it to be silent.

Reply:
Because these get returned to the users, it would be very easy to call destroy on them and they don't want their tool to come crashing down, an error would be fine (I suggested silent because then users dont have to consider this type of blockage when deleting all the blockages they created, not realizing that these are in the list too), but critical is too restrictive

Reply:
Not a warning, an error.  Critical is too severe and does not match what we do with dont-touch.

Reply:
A silent error is just a warning. So we need to decide what happens if a user tries delete a system blockage. Should it throw and stop mid way or keep going, ignoring the deletion request?

Reply:
I don't agree it should be silent.   It should just be a regular error (which is what dont-touch does).

Reply:
Error is fine with me